Firmware/MotorControl/encoder.cpp

```diff
@@ -257,7 +257,7 @@ bool Encoder::run_hall_polarity_calibration() {
             hall_polarity = 0b010;
         } else if (flip_detect(states, 3)) {
             hall_polarity = 0b100;
-        } else {
+        } else if (!config_.ignore_illegal_hall_state){
```

Comment:
Is this fallback to (the seemingly arbitrary) `hall_polarity = 0` really what we want? Maybe we can instead just change `uint8_t states = state_seen.to_ulong();` to `uint8_t states = state_confirmed.to_ulong();`.

Reply:
Yes polarity 0 is the "standard" polarity. There are lots of bug reports of the hall polarity detection and I think we should just give up on it on the old FW and fix it for the new one.